lexos/processors/analyze/content_analysis_model.py

```diff
@@ -0,0 +1,461 @@
+            count = 0
+            for phrase in dictionaries:
+                if phrase.dict_label == dictionary.label:
+                    count += phrase.count
+            counter.append(count)
+            if len(counter) == len(active_dicts):
+                self._counters.append(counter)
+
+    def generate_scores(self):
+        """calculate the formula and scores=formula/total_word_count for each
+        file in the corpus
+
+        """
+        self._scores = []
+        self._formulas = []
+        active_dicts = self.get_active_dicts()
+        result = 0
+        for i in range(len(self._corpus)):
+            new_formula = self._formula
+            for j in range(len(active_dicts)):
+                new_formula = new_formula.replace(
+                    "[" + active_dicts[j].label + "]",
+                    str(self._counters[i][j]))
+            new_formula = new_formula.replace("()", "")
+            try:
+                result = eval(new_formula)
+            except (ValueError, SyntaxError):
+                pass
+            self._scores.append(round(
+                float(result) / self._corpus[i].total_word_counts, 3))
+            self._formulas.append(result)
+
+    def generate_averages(self):
+        """Calculates the averages of eachm dictionary count, formula,
+        total_word_count, and score
+
+        """
+        self._averages = []
+        scores_sum = 0
+        total_word_counts_sum = 0
+        formulas_sum = 0
+        active_dicts = self.get_active_dicts()
+        for index, (score, formula, file) in enumerate(zip(self.scores,
+                                                           self._formulas,
+                                                           self._corpus)):
+            scores_sum += score
+            total_word_counts_sum += \
+                file.total_word_counts
+            formulas_sum += formula
+        if len(self.scores) != 0:
+            scores_avg = round(
+                (float(scores_sum) / len(self.scores)), 3)
+        else:
+            scores_avg = 0
+        if len(self._corpus) != 0:
+            total_word_counts_avg = round((float(total_word_counts_sum) /
+                                           (len(self._corpus))), 1)
+        else:
+            total_word_counts_avg = 0
+        if len(self._formulas) != 0:
+            sums_avg = round((float(formulas_sum) / len(self._formulas)), 1)
+        else:
+            sums_avg = 0
+        self._averages.append("Averages")
+        for x in range(len(active_dicts)):
+            cat_count = sum([counter[x] for counter in self._counters])
+            if len(self._counters) != 0:
+                self._averages.append(round(
+                    float(cat_count) / len(self._counters), 1))
+            else:
+                self._averages.append(0)
+        self._averages.append(sums_avg)
+        self._averages.append(total_word_counts_avg)
+        self._averages.append(scores_avg)
```

Comment:
You are appending to what was an empty list. I see no reason why you couldn't use list comprehension instead. Please do this wherever possible, instead of appending

Reply:
I'm not sure how to do that, inside the loop there is a line that always gets executed and an if/else statement

Reply:
I meant more that you could do a comprehension of sums_avg, total_word_counts_avg, etc, since these are always appended and I don't think the order they are in matters. You could then append your conditional averages to that